app/main/lib/shared_models/audio_model.py

```diff
@@ -212,14 +207,28 @@ def get_context_for_search(self, task):
         return context
 
     def search(self, task):
-        audio, temporary = self.get_audio(task)
-        context = self.get_context_for_search(task)
+        if "body" in task:
+            body = task.get("body", {})
+            threshold = task.get("raw", {}).get('threshold', 0.0)
+            limit = body.get("raw", {}).get("limit")
+        else:
+            body = task
+            threshold = body.get('threshold', 0.0)
+            limit = body.get("limit")
+        audio, temporary = self.get_audio(body)
+        context = self.get_context_for_search(body)
+        if audio.chromaprint_fingerprint is None:
+            callback_url =  Presto.add_item_callback_url(app.config['ALEGRE_HOST'], "audio")
+            if task.get("doc_id") is None:
+                task["doc_id"] = str(uuid.uuid4())
+            response = json.loads(Presto.send_request(app.config['PRESTO_HOST'], "audio__Model", callback_url, task).text)
+            result = Presto.blocked_response(response, "audio")
+            audio.chromaprint_fingerprint = result["response"]["hash_value"]
+            context = self.get_context_for_search(result["body"]["raw"])
         if audio:
-            threshold = task.get('threshold', 0.0)
             matches = self.search_by_hash_value(audio.chromaprint_fingerprint, threshold, context)
-            limit = task.get("limit")
             if temporary:
-                self.delete(task)
+                self.delete(body)
             if limit:
                 return {"result": matches[:limit]}
```

Comment:
If we take my comment above, then this line and the return below as well need to:
1. Call the callback URL to Check API if we have previously returned / called Presto
2. Simply return as we do if we have not.